pdns/dnsparser.cc

```diff
@@ -80,7 +80,7 @@ void UnknownRecordContent::toPacket(DNSPacketWriter& pw) const
   pw.xfrBlob(string(d_record.begin(),d_record.end()));
 }
 
-shared_ptr<DNSRecordContent> DNSRecordContent::deserialize(const DNSName& qname, uint16_t qtype, const string& serialized, uint16_t qclass)
+shared_ptr<DNSRecordContent> DNSRecordContent::deserialize(const DNSName& qname, uint16_t qtype, const string& serialized, uint16_t qclass, bool trusted)
```

Comment:
Small nit: I'm not a big fan of using `trusted` for this variable name, I'm pretty sure the next person reading this code will wonder what that means exactly. Perhaps something like `internalRepresentation`?

Reply:
It's (currently) `trusted` for "coming from a trusted source". I had also considered using a `bool` with a reverse polarity, which would be called `fromWire` - would you prefer that?

Reply:
Yeah, I think it would be better. What I'm worried about with `trusted` is that at some point someone interprets it as "coming from a trusted source, possibly over the network" which is not what we want. We could also write a comment, which might actually be a good idea, something like:
> fromWire is set to false when the content comes is deserialized from an internal source, like the LMDB backend.

Reply:
fromWire is tricky, as LMDB also is "wire format", but I think it at least is clearer than `trusted`. `internalRepresentation` is not too bad

Reply:
fromLMDB? Or is that too narrow?

Reply:
too narrow. LMDB is not even the only backend that uses wire storage (tinydns is the other one), and one day we might add even more.

Saying that out loud makes me realise that once a test is added, we might indeed find the same problem with tinydns - which then should be easy to fix.

Reply:
last bid: `fromBackend` :)

Reply:
I went with `internalRepresentation` which has two votes (we're outnumbered, Otto!)